src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -149,12 +143,80 @@ class TCFExperienceContents(
     tcf_system_relationships: List[TCFVendorRelationships] = []
 
 
-def get_matching_privacy_declarations(db: Session) -> Query:
-    """Returns flattened system/privacy declaration records where we have a matching gvl data use AND the
-    legal basis for processing is "Consent" or "Legitimate interests"
+def get_legal_basis_override_subquery(db: Session) -> Alias:
+    """Subquery that allows us to globally override a purpose's legal basis for processing.
+
+    Original legal basis for processing is returned where:
+    - feature is disabled
+    - declaration's legal basis is not flexible
+    - no legal basis override specified
+
+    Null is returned where:
+    - Purpose is excluded (this will effectively remove the purpose from the Experience)
+
+    Otherwise, we return the override!
+    """
+    if not CONFIG.consent.override_vendor_purposes:
+        return db.query(
+            PrivacyDeclaration.id,
+            PrivacyDeclaration.legal_basis_for_processing.label(
+                "overridden_legal_basis_for_processing"
+            ),
+        ).subquery()
 
-    Only systems that meet this criteria should show up in the TCF overlay.
+    return (
+        db.query(
+            PrivacyDeclaration.id,
+            case(
+                [
+                    (
+                        PrivacyDeclaration.flexible_legal_basis_for_processing.is_(
+                            False
+                        ),
+                        PrivacyDeclaration.legal_basis_for_processing,
+                    ),
+                    (
+                        TCFPurposeOverride.is_included.is_(False),
+                        None,
+                    ),
+                    (
+                        TCFPurposeOverride.required_legal_basis.is_(None),
```

Comment:
This is the new primary location where the purpose overrides are applied.

 Instead of using a hybrid property (which generated an awful SQL query, good call on checking that Adam), I'm joining with a subquery that has similar logic (and generates the SQL query I hoped the hybrid property would generate)

Reply:
nice, yeah this is still relatively easy to follow (maybe even easier?), especially considering the complex logic that you're encoding!